testsuite/cluster-test/src/tx_emitter.rs

```diff
@@ -400,22 +441,34 @@ impl SubmissionWorker {
                         // To avoid negative result caused by uncommitted tx occur
                         // Simplified from:
                         // end_time * num_committed - (tx_offset_time/num_requests) * num_committed
-                        (end_time - tx_offset_time / num_requests as u64) * num_committed as u64,
+                        // to
+                        // (end_time - tx_offset_time / num_requests) * num_committed
+                        latency * num_committed as u64,
                         Ordering::Relaxed,
                     );
+                    let latency_bucket_num = get_bucket_num(latency);
+                    {
+                        self.latencies.buckets[latency_bucket_num as usize]
```

Comment:
Probably makes sense to make `AtomicHistogram` a self contained struct (perhaps move it out of this file to some new file like `histogram.rs`.
And then AtomicHistogram should have some method like 'record_latency(latency_ms)` and buckets should not be exposed directly?